docs/building-apps/esm-views.md

```diff
@@ -0,0 +1,131 @@
+   and matched with their version in the package's or the package's root
+   `package.json`
+2. The main endpoint (as defined in the view's `package.json`'s `main` field)
+   and its local imports are bundled in a single file
+3. All the non-local dependencies encountered in the process are rewritten to
+   `import` from an external ESM CDN (by default https://www.skypack.dev/),
+   using the versions extracted in step 1
+4. All the local CSS is bundled in a single file
+5. The `dist` directory is generated, containing:
+   - the js bundle
+   - the css bundle
+   - a package manifest (`package.json`) file containing
+     ([see also below](#package-manifest)):
+     - the location of the js bundle (`"module"` field)
+     - the location of the css bundle (`"style"` field)
+     - an object with the whole set of dependencies and their version ranges
+       (`"dependencies"` field)
+     - an array of bundled dependencies (`bundledDependencies` field)
+   - a synthetically generated `index.html` file, linking the trampoline file
+     and the css bundle
+   - a synthetically generated trampoline file, dynamically `import`ing the js
+     bundle
+
+The view build result can either be
+[dynamically `import`ed](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports)
+from an host application
+([including the css bundle](https://web.dev/css-module-scripts/)) or served
+statically as a standalone application (for example, using `modular serve`).
+
+## Customise the ESM CDN
+
+A custom template for the CDN used when rewriting can be specified. The template
+is specified in the `EXTERNAL_CDN_TEMPLATE` environment variable and it is a
+string in which the string `[name]` is replaced with the name of the imported
+dependency and the string `[version]` is replaced with the version of the
+imported dependency, as extracted from the package's or the root's (hoisted)
+`package.json`
+
+For example:
+
+- The default template for the [Skypack](https://www.skypack.dev/) public CDN is
+  `EXTERNAL_CDN_TEMPLATE="https://cdn.skypack.dev/[name]@[version]"`
+- A valid template to work with the esm.sh public CDN can be specified with
+  `EXTERNAL_CDN_TEMPLATE="https://esm.sh/[name]@[version]"`
+
+## Customise bundling / rewriting strategy
+
+By default, all external dependencies are rewritten to a CDN url and none are
+bundled. This logic can be controlled using two environment variables:
+
+- `EXTERNAL_ALLOW_LIST` is a comma-separated string that specifies which
+  dependencies are allowed to be rewritten to CDN; if not specified, its default
+  value is '\*' ( -> all dependencies are rewritten)
+- `EXTERNAL_BLOCK_LIST` is a comma-separated string that specifies which
+  dependencies are **not** allowed to be rewritten to CDN; if not specified its
+  default value is '' ( -> no dependency excluded, i.e. all dependencies are
+  rewritten)
+
+The allow / block lists are parsed and processed according to this logic:
```

Comment:
Maybe change it to a numbered list to represent that we perform these checks in this particular order?